packages/magic/src/index.ts

```diff
@@ -0,0 +1,206 @@
+import type { WalletInit, APIKey, EIP1193Provider } from '@web3-onboard/common'
+
+function magic(options: APIKey): WalletInit {
+  const { apiKey } = options
+  const walletName = 'Magic Wallet'
+
+  return () => {
+    return {
+      label: walletName,
+      getIcon: async () => (await import('./icon.js')).default,
+      getInterface: async ({ EventEmitter, BigNumber, chains }) => {
+        const { Magic } = await import('magic-sdk')
+        const loginModal = (await import('./login-modal.js')).default
+        const brandingHTML = (await import('./branding.js')).default
+        let loggedIn: boolean = false
+
+        const {
+          createEIP1193Provider,
+          ProviderRpcErrorCode,
+          ProviderRpcError
+        } = await import('@web3-onboard/common')
+
+        const emitter = new EventEmitter()
+
+        let currentChain = chains[0].id
+
+        let customNodeOptions = {
+          chainId: parseInt(chains[0].id, 10),
```

Comment:
```suggestion
          chainId: parseInt(chains[0].id),
```
I had to fix this recently in other modules where I was using the radix incorrectly which causes the number to be parsed incorrectly.